docs/maintaining-dependencies.md

```diff
@@ -0,0 +1,258 @@
+        ],
+    )
+```
+
+Update the URL to the new tag:
+
+```text
+        urls = [
+            "https://github.com/open-telemetry/opentelemetry-proto/archive/v1.3.2.tar.gz",
+        ],
+```
+
+Update strip_prefix to the new tag:
+
+```text
+        strip_prefix = "opentelemetry-proto-1.3.2",
+```
+
+Download the new URL:
+
+```shell
+wget https://github.com/open-telemetry/opentelemetry-proto/archive/v1.3.2.tar.gz
+```
+
+Run a checksum on the new file:
+
+```shell
+sha256sum v1.3.2.tar.gz
+```
+
+```shell
+c069c0d96137cf005d34411fa67dd3b6f1f8c64af1e7fb2fe0089a41c425acd7  v1.3.2.tar.gz
+```
+
+Update the checksum in file bazel/repository.bzl:
+
+```text
+        sha256 = "c069c0d96137cf005d34411fa67dd3b6f1f8c64af1e7fb2fe0089a41c425acd7",
+```
+
+Typical change:
+
+```shell
+[malff@malff-desktop opentelemetry-cpp]$ git diff bazel/repository.bzl
+diff --git a/bazel/repository.bzl b/bazel/repository.bzl
+index bac1e45b..508b95a3 100644
+--- a/bazel/repository.bzl
++++ b/bazel/repository.bzl
+@@ -88,10 +88,10 @@ def opentelemetry_cpp_deps():
+         http_archive,
+         name = "com_github_opentelemetry_proto",
+         build_file = "@io_opentelemetry_cpp//bazel:opentelemetry_proto.BUILD",
+-        sha256 = "bed250ceec8e4a83aa5604d7d5595a61945059dc662edd058a9da082283f7a00",
+-        strip_prefix = "opentelemetry-proto-1.3.1",
++        sha256 = "c069c0d96137cf005d34411fa67dd3b6f1f8c64af1e7fb2fe0089a41c425acd7",
++        strip_prefix = "opentelemetry-proto-1.3.2",
+         urls = [
+-            "https://github.com/open-telemetry/opentelemetry-proto/archive/v1.3.1.tar.gz",
++            "https://github.com/open-telemetry/opentelemetry-proto/archive/v1.3.2.tar.gz",
+         ],
+     )
+```
+
+#### file cmake/opentelemetry-proto.cmake
+
+Update the tag in the CMake logic:
+
+```cmake
+        set(opentelemetry-proto "v1.3.2")
+```
+
+Typical change:
+
+```shell
+[malff@malff-desktop opentelemetry-cpp]$ git diff cmake/opentelemetry-proto.cmake
+diff --git a/cmake/opentelemetry-proto.cmake b/cmake/opentelemetry-proto.cmake
+index 19516c3b..dd6213c1 100644
+--- a/cmake/opentelemetry-proto.cmake
++++ b/cmake/opentelemetry-proto.cmake
+@@ -49,7 +49,7 @@ else()
+          "opentelemetry-proto=[ \\t]*([A-Za-z0-9_\\.\\-]+)")
+         set(opentelemetry-proto "${CMAKE_MATCH_1}")
+       else()
+-        set(opentelemetry-proto "v1.3.1")
++        set(opentelemetry-proto "v1.3.2")
+       endif()
+       unset(OTELCPP_THIRD_PARTY_RELEASE_CONTENT)
+     endif()
+```
+
+If opentelemetry-proto contains new files,
+the makefile needs to be adjusted to build the new code.
+
+#### file MODULE.bazel
+
+Make sure the new tag is available in bazel central:
+
+* [bazel-central](https://registry.bazel.build/modules/opentelemetry-proto)
+
+If missing, file a PR to add it, or contact the maintainer:
+
+* [repository](https://github.com/bazelbuild/bazel-central-registry/tree/main/modules/opentelemetry-proto)
+
```

Comment:
Can the upgrade proceed if the new tag is not available in bazel central? Not related to this PR, but trying to understand the role of bazel central here - Are we actually fetching the otel-proto depedency through bazel registry, or through repositories.bzl. cc @keith

Reply:
Oops, the intent is that bzlmod is running on CI atm, since users should be moving to that, we do want breakages of the bzlmod only build to break CI

Fixing here https://github.com/open-telemetry/opentelemetry-cpp/pull/2995

Reply:
If this was running correctly the answer is yes that this has to be added to the BCR first. if bzlmod wasn't blocking on CI we still wouldn't want to update the MODULE.bazel since any HEAD users would be broken by this

Reply:
Fixed the doc to reflect recent changes: CI does now depend on modules, and on bazel central.